spec/utils/file_transfer_spec.rb

```diff
@@ -0,0 +1,112 @@
+# frozen_string_literal: true
+
+require "utils/file_transfer"
+require "utils/file_transfer_error"
+require "spec_helper"
+require "services"
+
+# Since rclone does not care what the remote location is,
+# and since we do not have a separate test dropbox,
+# these tests use directories under /tmp/ as a fake remote location.
+#
+# So, CAVEAT EMPTOR, it does not truly test rclone's ability to
+# transfer files to a remote location, just that the wrapper code
+# (Utils::FileTransfer) does what it promises.
+
+RSpec.describe Utils::FileTransfer do
+  let(:ft) { described_class.new }
+  # Do not change local_dir or remote_dir to something important,
+  # as they WILL be deleted during these tests.
+  let(:local_dir) { "/tmp/file_transfer_test/local" }
+  let(:remote_dir) { "/tmp/file_transfer_test/remote" }
+  let(:local_file_name) { "test_file_a.txt" }
+  let(:remote_file_name) { "test_file_b.txt" }
+  let(:local_file_path) { "#{local_dir}/#{local_file_name}" }
+  let(:remote_file_path) { "#{remote_dir}/#{remote_file_name}" }
+  let(:conf) { "config/rclone.conf" }
+  let(:missing_dir) { "/dev/null/foo" }
+  let(:missing_file) { "/dev/null/foo.txt" }
+
+  # Clean up temporary files and directories
+  before(:each) do
+    Settings.rclone_config_path = conf
```

Comment:
I think we can probably avoid setting this here and just do it in `settings/test.yml`. If we do change it here, we should change it back to the previous value, e.g:

```ruby
before(:each) do
  @old_rclone_config_path = Settings.rclone_config_path
  Settings.rclone_config_path = "/mashed/potatoes"
end

after(:each) do
  Settings.rclone_config_path = @old_rclone_config_path
end
```

Reply:
I cannot get this to work and it's really bugging me, but I have to quit for the day.

Reply:
yup